src/Sulu/Bundle/PageBundle/Resources/js/views/Form/toolbarActions/TemplateToolbarAction.js

```diff
@@ -26,6 +27,23 @@ export default class TemplateToolbarAction extends AbstractFormToolbarAction {
             this.resourceFormStore.setType(this.defaultTemplate);
         }
 
+        const parentPageId = this.router.attributes.parentId;
+        if (parentPageId && this.webspace && !this.parentPage) {
+            ResourceRequester.get('pages', {
+                id: parentPageId,
+                language: this.router.attributes.locale,
+            }).then((response) => {
+                this.parentPage = response;
+                const parentTemplate = response.template;
+                for (const defaultTemplate of this.webspace.defaultTemplates['page']) {
```

Comment:
Instead of looping over everything I would use the [`find` function of the array](https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/Array/find), which results in less levels of nesting.

Reply:
Replace with the find function